contracts/types/WeightedSigners.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/**
+ * @notice This struct represents the weighted signers payload
+ * @param signers The list of signers
+ * @param weights The list of weights
+ * @param threshold The threshold for the signers
+ */
+struct WeightedSigner {
+    address signer;
+    uint128 weight;
```

Comment:
`uint128` is more commonly supported in cosmwasm and other Rust VMs so switching to that. Should still be sufficient for us.

Reply:
We should use `uint96` to fit signer and weight into a one storage slot.

Reply:
It's fine if other implementations will use 128 on their site.

Reply:
we don't store WeightSigners, only the hash